### PR TITLE
update nvfuser install instructions

### PR DESCRIPTION
**Pull request by t-vi**
This is what I like to use.
cc @xwang233 @jjsjann123 
also cc @nikitaved

### CODE REVIEW

File: docs/source/fundamentals/installation.rst [hunk 1]
```diff
@@ -6,11 +6,11 @@ Minimal dependencies
 
 Follow these instructions to install PyTorch, nvFuser, and finally Thunder.
 
-Install PyTorch and nvFuser with pip (command shown is for CUDA 12.1)::
+Install PyTorch and nvFuser with pip (command shown is for CUDA 12.1 and PyTorch 2.4.x)::
```
[the remark this answers is not in the document]
Comment (t-vi) [reply docs/source/fundamentals/installation.rst:9]: My understanding is that 121 is for CUDA 12.x with x >= 1

[the remark this answers is not in the document]
Comment (nikitaved) [reply docs/source/fundamentals/installation.rst:9]: These are not available, apparently. I tried to install wheels for CUDA 12.5

File: docs/source/fundamentals/installation.rst [hunk 2]
```diff
@@ -6,11 +6,11 @@ Minimal dependencies
...
 
-  pip install --pre "nvfuser-cu121[torch]" --extra-index-url https://pypi.nvidia.com
+  pip install --pre nvfuser-cu121-torch24
 
-cu121 can be replaced with cu118 depending on your CUDA version.
+cu121 can be replaced with cu118 depending on your CUDA version. NVFuser builds typically support the latest point release of PyTorch stable versions.
```
Comment (nikitaved) [inline docs/source/fundamentals/installation.rst:13]: I think we can point to `https://pypi.nvidia.com` as it has wheels for versions other than 11.8 and 12.1.

Comment (nikitaved) [reply docs/source/fundamentals/installation.rst:13]: Also, if I follow the link, I see versions with torch and without torch. What is the difference? Which one should I choose?

File: README.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ To use Thunder on your local machine:
 
 ```bash
 # install nvFuser which installs the matching nightly PyTorch
-pip install --pre 'nvfuser-cu121[torch]' --extra-index-url https://pypi.nvidia.com
+pip install --pre nvfuser-cu121-torch24
```
Comment (xwang233) [inline README.md:82]: `nvfuser-...-torch24` builds against stable release of torch 2.4. I think we also need to update the line above "which installs the matching ~~nightly~~ stable"

File: docs/source/fundamentals/installation.rst (hunk 2 above)
Comment (xwang233) [reply docs/source/fundamentals/installation.rst:13]: torch 2.3 has cuda 11.8 and 12.1
torch 2.4 has cuda 11.8, 12.1, and 12.4

nvfuser nightly on https://pypi.nvidia.com has the cuda matrix (currently the same as torch 2.4)

Comment (t-vi) [reply docs/source/fundamentals/installation.rst:13]:
```suggestion
cu121 can be replaced with cu118 depending on your CUDA version. NVFuser builds typically support the latest point release of PyTorch stable versions.
For torch 2.4, cu124 is also supported. For nightly versions and more detailed instructions, please see https://github.com/NVIDIA/Fuser/#installation
```

**Review by lantiga — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (t-vi) [reply README.md:82]: Yes, thank you!

File: docs/source/fundamentals/installation.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (t-vi) [reply docs/source/fundamentals/installation.rst:9]: Let's go with this until the next person improves further. :)

**Pull request merged**